wisp-resource-loader/src/main/kotlin/wisp/resources/ClasspathResourceLoaderBackend.kt

```diff
@@ -83,13 +83,15 @@ object ClasspathResourceLoaderBackend : ResourceLoader.Backend() {
     }
 
     override fun open(path: String): BufferedSource? {
-        val classLoader = Thread.currentThread().contextClassLoader
-        val resourceAsStream = classLoader.getResourceAsStream(path.removePrefix("/")) ?: return null
+        val resourceAsStream = classLoader().getResourceAsStream(path.removePrefix("/")) ?: return null
         return resourceAsStream.source().buffer()
     }
 
     override fun exists(path: String): Boolean {
-        val classLoader = Thread.currentThread().contextClassLoader
-        return classLoader.getResource(path.removePrefix("/")) != null
+        return classLoader().getResource(path.removePrefix("/")) != null
+    }
+
+    private fun classLoader() : ClassLoader {
+        return Thread.currentThread().contextClassLoader ?: ClassLoader.getSystemClassLoader()
```

Comment:
Maybe good to have a comment on when `contextClassLoader` can be null?

Reply:
TBH I'm not really sure under exactly what conditions it occurs. I'm seeing it happen when running from an armeria service handler implementation but I don't know why that's so special. According to the javadoc for `contextClassLoader` it should fall back to the parent thread if the current thread doesn't have one, so my best guess is that the armeria code is actually setting it to null explicitly.